test/test_basic.py

```diff
@@ -94,7 +94,7 @@ async def test_connection_refused(runtime_dir: pathlib.Path) -> None:
 @pytest.mark.asyncio
 async def test_dns_error(runtime_dir: pathlib.Path) -> None:
     session = ferny.Session()
-    with pytest.raises(socket.gaierror):
+    with pytest.raises((socket.gaierror, ferny.ssh_errors.SshError)):
```

Comment:
follow-ups:

 - is it worth capturing and categorizing the new message?  I don't like that we blanket-capture SshError here.
 - should we try another hostname that is sure to get us a gaierror without hanging in case there's no network access?

Reply:
Indeed, sent PR #26 for these